packages/rest/test/resource.test.ts

```diff
@@ -160,26 +160,38 @@ describe("rest: resources", () => {
         }
 
         @post
-        @collectionAction(Thing, "export")
+        @collectionAction(Thing, "export1")
         op exportThing(): {};
 
+        #suppress "deprecated"
         @post
-        @collectionAction(Thing, "export")
+        @collectionAction(Thing, "export2")
         @segmentSeparator(":")
-        op exportThingWithColon(): {};
+        op exportThingWithColon1(): {};
+
+        @post
+        @collectionAction(Thing, "export3")
+        @actionSeparator(":")
+        op exportThingWithColon2(): {};
       }
       `
     );
 
     deepStrictEqual(routes, [
       {
         verb: "post",
-        path: "/things/export",
+        path: "/things/export1",
+        params: [],
+      },
+      {
+        verb: "post",
+        // FIXME: This is the actual correct behavior for segment separator... but it is breaking...
+        path: "/:things:export2",
```

Comment:
This is exactly the problem with `@segmentSeparator`. Because this gets evaluated at the end, this is what happens. I don't know how we should "fix" this (or if we should). cc/ @daviwil

Reply:
I wouldn't worry about this case, it looks wrong to begin with.